### PR TITLE
Gnome 45 compatibility

### PR DESCRIPTION
**Pull request by kenohassler**
This PR contains the changes to imports required for Gnome 45. Since **it breaks backwards compatibility**, I created a new "legacy" branch from master before this PR that contains the old code compatible with versions 40-44.

I tried to keep the changes minimal, but it was still necessary to change every JS file, since all imports had to be updated.
This addresses issue #212.

### CODE REVIEW

**Comment by derenv** [comment]
The "legacy" branch seems a reasonable accommodation for anyone not yet using gnome 45 - we should probably update the README to make it clear though. There are a couple other updates needed there for similar reasons, i.e. #202, so we could leave that for a later documentation-focused update instead.

I'll review and test this in a bit more detail at some point this/next week, but having had a brief look at the changes they seem good/reasonable!

**Comment by kenohassler** [comment]
yeah good point, the readme is a bit dated. I opened PR #214 to address these issues, it might need to be rebased before merging.

**Review by derenv — COMMENTED** [review]
This looks good! I need to test this in a Gnome VM asap, then will approve review if there are no issues.

**Review by derenv — APPROVED** [review]
Appears to work as expected in test VM (Ubuntu 23.10.1, both Wayland/X11)!

**Pull request merged**
